### ISSUE TITLE
BUG: History.push() allows navigation with prompt

### ISSUE THREAD

**Issue by Nikodermus** [ZacharyRSmith/react-router-navigation-prompt#49]
"react-router-navigation-prompt": "^1.8.5",
## BUG

Redirection made through history.push() is not taken in account for the prompt to be triggered and allows redirection in the end.

### Steps to reproduce:
* Given a component which has access to the history obj (Like using `withRouter` HOC)
* Create a prompt with a `when={true}`
* A button which `onClick={()=> props.history.push('/')}`
* It will redirect without triggering the prompt

### Expected behavior:
Prompt should be triggered even when the router change is made with the `history.push` function.

The button click resolves a redux action which then triggers the history.push(), maybe it has to do?

### Additional Info

```
├─┬ docz@0.13.7
│ ├─┬ docz-core@0.13.7
│ │ └─┬ @mdx-js/loader@0.16.8
│ │   └─┬ @mdx-js/tag@0.16.8
│ │     └── react@16.8.3  extraneous
│ ├── react@16.8.3 
│ ├── react-router@4.3.1 
│ └─┬ react-router-dom@4.3.1 
│   └── react-router@4.3.1  deduped
├─┬ docz-theme-default@0.13.7
│ ├─┬ @emotion/styled@10.0.7
│ │ └─┬ @emotion/styled-base@10.0.7
│ │   └── react@16.8.3  extraneous
│ ├── react@16.8.3 
│ └─┬ react-live@1.12.0
│   └── react@16.8.3  extraneous
├─┬ docz-theme-hydrate@1.5.1
│ ├─┬ docz@0.12.17
│ │ ├── react@16.8.3  deduped
│ │ ├── react-router@4.3.1  deduped
│ │ └─┬ react-router-dom@4.3.1 
│ │   └── react-router@4.3.1  deduped
│ └── react@16.8.3 
├── react@16.6.0 
├─┬ react-ga@2.5.3
│ └── react@16.8.3 
├─┬ react-router-dom@4.2.2 
│ └── react-router@4.3.1  deduped
└── react-router-navigation-prompt@1.8.5 
```

**Comment by ZacharyRSmith** [comment]
Thanks for the issue, @Nikodermus .  Unfortunately, I'm not sure when I'll be able to look at it.  Would you please PR a fix?

**Comment by Pau1fitz** [comment]
@Nikodermus @ZacharyRSmith  I am seeing the same, I think it is only when you push to `'/'` -  `props.history.push('/')`, at least this is the only route I am seeing this behaviour. 

If I do `props.history.push('/'test)` there is no problem.

Just investigating now, if I find the fix I will create a PR.

**Comment by Pau1fitz** [comment]
@Nikodermus @ZacharyRSmith  I am seeing the same, I think it is only when you push to `'/'` -  `props.history.push('/')`, at least this is the only route I am seeing this behaviour. 

If I do `props.history.push('/'test)` there is no problem. @Nikodermus can you check if it works if you don't use `'/'` but instead navigate to a different route?

**Comment by Nikodermus** [comment]
Indeed, this is working fine with routes different to '/', do you maybe a baseName different to '/' in your app @Pau1fitz ?

**Comment by Joachimzeelmaekers** [comment]
@Nikodermus Could you try using 
`<NavLink to="/">
    Back to overview
  </NavLink>`
component instead of a button?

**Comment by ohelixa** [comment]
Any news on this?
